shared/misc/upload_ansible_roles_to_galaxy.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python2
+
+from __future__ import print_function
+
+from tempfile import mkdtemp
+import os
+import os.path
+import sys
+import shutil
+import re
+import argparse
+import getpass
+import yaml
+
+try:
+    from github import Github, InputGitAuthor
+except ImportError:
+    sys.stderr.write("Please install PyGithub, on Fedora it's in the "
+                     "python-PyGithub package.\n")
+    sys.exit(1)
+
+
+ORGANIZATION_NAME = "Ansible-Security-Compliance"
+GIT_COMMIT_AUTHOR_NAME = "SCAP Security Guide development team"
+GIT_COMMIT_AUTHOR_EMAIL = "scap-security-guide@lists.fedorahosted.org"
+META_TEMPLATE_PATH = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)),
+    "ansible_galaxy_meta_template.yml"
+)
+README_TEMPLATE_PATH = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)),
+    "ansible_galaxy_readme_template.md"
+)
+
+
+def create_empty_repositories(github_new_repos, github_org):
+    for github_new_repo in github_new_repos:
+        print("Creating new Github repository: %s" % github_new_repo)
+        github_org.create_repo(
+            github_new_repo,
+            description="Role generated from SCAP Security Guide",
+            homepage="https://www.open-scap.org/",
+            private=False,
+            has_issues=False,
+            has_wiki=False,
+            has_downloads=False)
+
+
+def clone_and_init_repository(parent_dir, repo):
+    os.system(
+        "git clone git@github.com:%s/%s.git" % (ORGANIZATION_NAME, repo))
+    os.system("ansible-galaxy init " + repo + " --force")
+    os.chdir(repo)
+    try:
+        os.system('git add .')
+        os.system('git commit -a -m "Initial commit" --author "%s <%s>"'
+                  % (GIT_COMMIT_AUTHOR_NAME, GIT_COMMIT_AUTHOR_EMAIL))
+        os.system('git push origin master')
+    finally:
+        os.chdir("..")
+
+
+def update_repository(repository, local_file_path):
+    print("Processing %s..." % repository.name)
+
+    with open(local_file_path, 'r') as f:
+        filedata = f.read()
+
+    role_data = yaml.load(filedata)
+    vars_data = []
+    if "vars" in role_data[0]:
+        vars_data = role_data[0]["vars"]
+
+    tasks_data = []
+    if "tasks" in role_data[0]:
+        tasks_data = role_data[0]["tasks"]
+
+    # TODO: what do we do about pre_tasks? ansible language doesn't allow it
+    #       for roles
+
+    tasks_local_content = yaml.dump(tasks_data, width=120, indent=4,
+                                    default_flow_style=False)
```

Comment:
Not a big deal, but the tasks lost the newline separation between them.
I don't know if this can be fixed.

Reply:
I was looking into that and I can't figure out how to fix that :-( Also the keys are reordered alphabetically which sucks but can't easily be fixed.